test/Interop/Cxx/constructor/Inputs/basic.h

```diff
@@ -0,0 +1,36 @@
+#ifndef TEST_INTEROP_CXX_CONSTRUCTOR_INPUTS_BASIC_H
+#define TEST_INTEROP_CXX_CONSTRUCTOR_INPUTS_BASIC_H
+
+struct HasUserProvidedCopyConstructor {
+  int value;
+  HasUserProvidedCopyConstructor(int val) : value(val) {}
+  HasUserProvidedCopyConstructor(HasUserProvidedCopyConstructor const &other)
+      : value(other.value + 1) {}
+};
+
+struct EmptyCopyConstructor {
```

Comment:
Hmm. I suppose we might not need it. But I don't think it will be much (if any) work to maintain and there is a slight difference. We don't want to, for instance, explode this and call two constructors (which is unlikely to happen but, why not double-check?).

Reply:
By "explode this and call two constructors", do you mean a situation where Swift would call both the copy constructor of `EmptyCopyConstructor` and `HasUserProvidedCopyConstrutor`? That's something that the other cases below (e.g. `HasNonTrivialDefaultCopyConstructor`) should cover too though?

Anyway, I'm not strongly opposed to keeping this in, it just seemed a bit redundant to me.

Reply:
No, you're right. This is redundant.